Sources/Swift/Persistence/SentryScopePersistentStore.swift

```diff
@@ -199,24 +190,19 @@ extension SentryScopePersistentStore {
 // MARK: - User
 extension SentryScopePersistentStore {
     private func encode(user: User) -> Data? {
-        guard let sanitizedUser = sentry_sanitize(user.serialize()) else {
-            SentrySDKLog.error("Failed to sanitize user, reason: user is not valid json: \(user)")
-            return nil
-        }
-        guard let data = SentrySerialization.data(withJSONObject: sanitizedUser) else {
+        guard let data = SentrySerialization.data(withJSONObject: user.serialize()) else {
             SentrySDKLog.error("Failed to serialize user, reason: user is not valid json: \(user)")
             return nil
         }
         return data
     }
     
     private func decodeUser(from data: Data) -> User? {
-        do {
-            let decoder = JSONDecoder()
-            return try decoder.decode(UserDecodable.self, from: data)
-        } catch {
-            SentrySDKLog.error("Failed to decode user, reason: \(error)")
-            return nil
-        }
+        return decoderAux(data)
+    }
+    
+    // Swift compiler can't infer T, even if I try to cast it
+    private func decoderAux(_ data: Data) -> UserDecodable? {
```

Comment:
`m`: I'm a bit confused by the name `decodeerAux`. What does it stand for?

Reply:
Ah OK, thanks. `decodeUserToWorkWithSwiftCompiler`, `decodeUserInternal`, `decodeUserHelper`, `decodeUserAuxiliary` all would work for me. `decoderAux` is a bit confusing. The comment above states it anyway. Pick whatever you prefer. I have no strong opinion.